lua/nvim-tree/explorer/filters.lua

```diff
@@ -4,14 +4,23 @@ local FILTER_REASON = require("nvim-tree.enum").FILTER_REASON
 local Class = require("nvim-tree.classic")
 
 ---@alias FilterType "custom" | "dotfiles" | "git_ignored" | "git_clean" | "no_buffer" | "no_bookmark"
+---@alias GitFilterType "git_clean" | "git_ignored"
+
+---@class FilterStatus
+---@field project GitProject | nil
+---@field bufinfo table
+---@field bookmarks table
 
 ---@class (exact) Filters: Class
 ---@field enabled boolean
 ---@field state table<FilterType, boolean>
+---@field api table<string, fun(self: Filters, path: string): boolean|nil>
 ---@field private explorer Explorer
 ---@field private exclude_list string[] filters.exclude
 ---@field private ignore_list table<string, boolean> filters.custom string table
 ---@field private custom_function (fun(absolute_path: string): boolean)|nil filters.custom function
+---@field protected status FilterStatus
+---@field private filter_cache table<string, FILTER_REASON>
```

Comment:
```suggestion
---@field private filter_cache table<string, FILTER_REASON>
---@field private custom      fun(self: Filters, path: string): boolean
---@field private dotfile     fun(self: Filters, path: string): boolean
---@field private git_ignored fun(self: Filters, path: string): boolean
---@field private git_clean   fun(self: Filters, path: string): boolean
---@field private buf         fun(self: Filters, path: string): boolean
---@field private bookmark    fun(self: Filters, path: string): boolean
```

This is annoying, but it will keep the language server from complaining about inject-field

You could even wrap them in the constructor, although I'm not sure how clean that would look.